frame/contracts/src/wasm/runtime.rs

```diff
@@ -1217,13 +1178,13 @@ define_env!(Env, <E: Ext>,
 	seal_deposit_event(ctx, topics_ptr: u32, topics_len: u32, data_ptr: u32, data_len: u32) => {
 		let num_topic = topics_len
 			.checked_div(sp_std::mem::size_of::<TopicOf<E::T>>() as u32)
-			.ok_or_else(|| ctx.store_err("Zero sized topics are not allowed"))?;
+			.ok_or_else(|| "Zero sized topics are not allowed")?;
```

Comment:
why not using a new pallet error here instead ?

Reply:
The reason I am hesitant about doing so is that this more of a programming error (assert) than a runtime error. Having a zero sized topic is a static failure of having a borked `Hash`:
```rust
pub type TopicOf<T> = <T as frame_system::Config>::Hash;
```

I cannot use `static_assertions` because the size depends on `T`. I am also hesitant to to use a plain `assertion` because of the late panic. This was somehow the middle ground.

Maybe there is a better place to assert such assumptions?

Reply:
I see, I read a bit too fast and thought it was about inputs. Then yes better not declaring another error. I think it is fine as it is now.